lm_eval/models/huggingface.py

```diff
@@ -1304,9 +1305,20 @@ def apply_chat_template(self, chat_history: List[Dict[str, str]]) -> str:
         """
         Method to apply a chat template to a list of chat history between user and model.
         """
-        return self.tokenizer.apply_chat_template(
-            chat_history, tokenize=False, add_generation_prompt=True
-        )
+        try:
+            chat_templated = self.tokenizer.apply_chat_template(
+                chat_history, tokenize=False, add_generation_prompt=True
+            )
+        except jinja2.exceptions.TemplateError:
+            eval_logger.warning(
+                "Failed to apply chat template. removing the system role in chat history."
+            )
+            chat_history = [msg for msg in chat_history if msg["role"] != "system"]
```

Comment:
What about appending it to the first `user` entry? That way the model still receives the full prompt we want to evaluate on, but the formatting is model dependent (same as chat template, tokenization, context length, etc.).

Reply:
The system prompt is usually an instruction to further guide the model. It should not contain the actual prompt for the task. We therefore went with the assumption that if it was not supported by the model we would just remove it and let the task prompt be the first user entry

Reply:
hmm. Maybe. But we do now default to using the task description as the system prompt: https://github.com/EleutherAI/lm-evaluation-harness/blob/058cfd0eeb022c0bc4862651a3ae08e4e046a106/lm_eval/api/task.py#L1068

And some tasks don't make sense without this context:
https://github.com/EleutherAI/lm-evaluation-harness/blob/058cfd0eeb022c0bc4862651a3ae08e4e046a106/lm_eval/tasks/leaderboard/bbh_mc/boolean_expressions.yaml#L2

Reply:
@baberabb is right that for some tasks (including MMLU, as a canonical example) a description includes pretty important info!

I think at minimum a warning-level log stating that system prompts not being supported may have adverse impacts on tasks' `description` fields is appropriate.

Or can we have several `"role": "user"` messages in a row? How about if there is a non-empty `description` value, we end up with

```
[
	{"role": "user", "content": f"{description}"},
	{"role": "user", "content":, {first example doc_to_text},
	...
]
```
instead of sticking the description into the sysprompt?

Reply:
> Or can we have several `"role": "user"` messages in a row? How about if there is a non-empty `description` value, we end up with

This wouldn't work as some (all?) chat templates — such as the one mistral uses — require successive roles to alternate between `user` and `assistant`